### ISSUE TITLE
implimente the `BCELoss` loss

### ISSUE THREAD

**Issue by JamorMoussa** [JamorMoussa/NanoTorch#21]
#### Description:
The `nanotorch` library is currently being tested for regression problems. While we have implemented the `MSELoss`, we now need to implement a `nn.BCELoss` for handling binary classification tasks.

#### Additional Information:
PyTorch documentation for `BCELoss`: [https://pytorch.org/docs/stable/generated/torch.nn.BCELoss.html](https://pytorch.org/docs/stable/generated/torch.nn.BCELoss.html)
